specification/cognitiveservices/OpenAI.Inference/models/azure.cognitive.search.options.tsp

```diff
@@ -0,0 +1,104 @@
+  contentFieldSeparator?: string;
+
+  @doc("The names of fields that represent vector data.")
+  @projectedName("csharp", "VectorFieldNames")
+  vectorFields?: string[];
+}
+
+@doc("""
+The type of Azure Cognitive Search retrieval query that should be executed when using it as an Azure OpenAI chat extension.
+""")
+@added(ServiceApiVersions.v2023_08_01_Preview)
+enum AzureCognitiveSearchQueryType {
+  @doc("Represents the default, simple query parser.")
+  simple: "simple",
+
+  @doc("Represents the semantic query parser for advanced semantic modeling.")
+  semantic: "semantic",
+
+  @doc("Represents vector search over computed data.")
+  vector: "vector",
+
+  @doc("Represents a combination of the simple query strategy with vector data.")
+  vectorSimpleHybrid: "vectorSimpleHybrid",
+
+  @doc("Represents a combination of semantic search and vector data querying.")
+  vectorSemanticHybrid: "vectorSemanticHybrid",
+}
+
+@added(ServiceApiVersions.v2023_08_01_Preview)
+@doc("""
+A specific representation of configurable options for Azure Cognitive Search when using it as an Azure OpenAI chat
+extension.
+""")
+model AzureCognitiveSearchChatExtensionConfiguration {
+  @doc("""
+  The type label to use when configuring Azure OpenAI chat extensions. This should typically not be changed from its
+  default value for Azure Cognitive Search.
+  """)
+  type: AzureChatExtensionType.azureCognitiveSearch;
+
+  @doc("The absolute endpoint path for the Azure Cognitive Search resource to use.")
+  @projectedName("csharp", "SearchEndpoint")
+  endpoint: url;
+
+  @doc("The API key to use with the specified Azure Cognitive Search endpoint.")
+  @projectedName("csharp", "SearchKey")
+  key: string;
+
+  @doc("The name of the index to use as available in the referenced Azure Cognitive Search resource.")
+  indexName: string;
+
+  @doc("Customized field mapping behavior to use when interacting with the search index.")
+  @projectedName("csharp", "FieldMappingOptions")
+  fieldsMapping?: AzureCognitiveSearchIndexFieldMappingOptions;
+
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "this represents the case-sensitive wire format"
+  @doc("The configured top number of documents to feature for the configured query.")
+  @projectedName("csharp", "DocumentCount")
+  topNDocuments?: int32;
+
+  @doc("The query type to use with Azure Cognitive Search.")
+  queryType?: AzureCognitiveSearchQueryType;
+
+  @doc("Whether queries should be restricted to use of indexed data.")
+  @projectedName("csharp", "ShouldRestrictResultScope")
+  inScope?: boolean;
+
+  @doc("The additional semantic configuration for the query.")
+  semanticConfiguration?: string;
+
+  @doc("When using embeddings for search, specifies the resource URL from which embeddings should be retrieved.")
+  embeddingEndpoint?: url;
+
+  @doc("When using embeddings, specifies the API key to use with the provided embeddings endpoint.")
+  embeddingKey?: string;
```

Comment:
We exposed search filter in recent release. Shall we add optional string field "filter" to align with service behavior? If it is already too late, I'm fine to add it in the next SDK release..

Reply:
We are in a bit of a time crunch at the moment. I will create an issue in our backlog to address this feedback. Thank you!